CMakeLists.txt

```diff
@@ -25,11 +25,20 @@ target_compile_definitions( DemoGafferExtensionModule PRIVATE BOOST_SIGNALS_NO_D
 target_link_libraries( DemoGafferExtensionModule DemoGafferExtension )
 
 if(APPLE)
-  target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/lib/Python.framework/Versions/2.7/include/python2.7 )
+  if( "${PYTHON_VERSION}" MATCHES "3.7" )
+    target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/lib/Python.framework/Versions/3.7/include/python3.7 )
+  else()
+    target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/lib/Python.framework/Versions/2.7/include/python2.7 )
+  endif()
   target_link_libraries( DemoGafferExtensionModule GafferBindings IECorePython boost_python Python)
 else()
-  target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/include/python2.7 )
-  target_link_libraries( DemoGafferExtensionModule GafferBindings IECorePython boost_python27 python2.7)
+  if( "${PYTHON_VERSION}" MATCHES "3.7" )
+    target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/include/python3.7m )
+    target_link_libraries( DemoGafferExtensionModule GafferBindings IECorePython boost_python37 python3.7m)
+  else()
+    target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/include/python2.7 )
+    target_link_libraries( DemoGafferExtensionModule GafferBindings IECorePython boost_python27 python2.7)
+  endif()
```

Comment:
As above, but maybe a bit trickier... I'm no cmake expert, but I'd hope there is a way to strip the `.` from ${PYTHON_VERSION}